app/src/main/java/ar/com/wolox/android/example/ui/training/homePage/news/NewsFragment.kt

```diff
@@ -0,0 +1,49 @@
+package ar.com.wolox.android.example.ui.training.homePage.news
+
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.widget.Toast
+import ar.com.wolox.android.R
+import ar.com.wolox.android.example.model.News
+import ar.com.wolox.wolmo.core.fragment.WolmoFragment
+import kotlinx.android.synthetic.main.news_fragment_page.*
+import javax.inject.Inject
+
+/**
+ * NewsFragment is used for display news
+ */
+class NewsFragment @Inject constructor() : WolmoFragment<NewsPresenter>(), INewsView {
+
+    private lateinit var mRecyclerView: RecyclerView
+    private lateinit var viewAdapter: RecyclerView.Adapter<*>
+    private lateinit var viewManager: RecyclerView.LayoutManager
+    private lateinit var news: MutableList<News>
+
+    override fun layout(): Int = R.layout.news_fragment_page
+
+    override fun init() {
+        news = presenter.loadNews()
```

Comment:
In the `MVP` pattern, the `view` must not know anything about when to work. The `view` only tell events to the `presenter` and receive orders by it. So, here it's not okay. The `view` don't know when to load the news, this `List` will start empty. So, to manage with `MVP` the steps should be something like this:

- The view initialized the `news` as an empty list.
- On the `presenter.onViewAttached()`:
   - The `presenter` get the news (if them are hardcoded, skip this point).
   - If there are more news to load:
      - The `presenter` send the news to the `view` by calling a method like you have but sending the news `addNews(news: News)`.
      - The `view` add the news received to the list. 
   - If there are not more news to load:
      - The `presenter` call a method of the `view` like `showNoMoreNews()`
      - The `view` show a toast with the error message.
- On the `view` on the `setOnRefreshListener` the `view` will only inform to the presenter that the news list is refreshed and the presenter will manage the same behaviour as in the `onViewAttached`.

I know it hard to understand, so if you have any doubt, talk to me.

Reply:
This is missing too!